### PR TITLE
re-wrote functions and all the tests, added dependencies

### LINKED ISSUE #19: write function and tests for plot_roc

### PR DESCRIPTION
**Pull request by taohuang-ubc**

### CODE REVIEW

**Comment by taohuang-ubc** [comment]
this should resolve #19

File: R/plot_roc.R [hunk 1]
```diff
@@ -3,19 +3,52 @@
 #' Takes in a vector of prediction labels and a vector of prediction probabilities.
 #'
 #' @param y_label a vector contains the validation set label
-#' @param predict_prob a vector contains the prediction probabilities
+#' @param predict_proba a vector contains the prediction probabilities
```
Comment (flizhou) [inline R/plot_roc.R]:
```suggestion
#' @param predict_proba a numeric vector contains the prediction probabilities
```

File: R/plot_roc.R [hunk 2]
```diff
@@ -3,19 +3,52 @@
 #' Takes in a vector of prediction labels and a vector of prediction probabilities.
 #'
 #' @param y_label a vector contains the validation set label
```
Comment (flizhou) [inline R/plot_roc.R]:
```suggestion
#' @param y_label a numeric vector contains the validation set label
```

File: tests/testthat/test-plot_roc.R [hunk 3]
```diff
@@ -1,3 +1,72 @@
-test_that("multiplication works", {
-  expect_equal(2 * 2, 4)
+library(ggplot2)
+library(plotROC)
+library(pROC)
+library(testthat)
```
Comment (flizhou) [inline tests/testthat/test-plot_roc.R]: You can add those packages as dependencies and remove those lines.

**Review by flizhou — COMMENTED** [review]
Please review my suggestions and revise test-plot_roc.R

**Comment by taohuang-ubc** [comment]
fanli, I applied all your changes locally, added all the packages and pushed again, should be fine now. Thanks.

**Review by flizhou — APPROVED** [review]
Looks good now, thanks!

**Pull request merged**